### PR TITLE
remove const qualifier if required

### PR DESCRIPTION
**Pull request by axdank**
fix:

image

### CODE REVIEW

**Comment by natecraddock** [comment]
@VisenDev what do you think of these changes?

**Comment by VisenDev** [comment]
I generally oppose using const cast unless completely necessary. However, from what I can tell it does seem to be legal behavior to mutate the result of `lua_tolstring`. 

If mutating the result of a `lua_tolstring` is defined behavior, then perhaps the correct this to do here is remove the `const` qualifier from this function
```zig
    pub fn toString(lua: *Lua, index: i32) ![:0]const u8 {
```
Rather than const casting its result.

**Comment by natecraddock** [comment]
What I read on this page https://www.lua.org/manual/5.4/manual.html#lua_tolstring (look for section 4.1.3), it is safe to mutate the result of `lua_tolstring` but only for a short time (roughly within a function call if I'm reading things right).

I think I'd rather allow the caller to use `@constCast()` as needed, or provide another function that returns a `[]u8` with a comment about when it is safe to use.

**Comment by natecraddock** [comment]
@axdank do you have any thoughts? Based on the lua reference manual 4.1.3, this could be a possible footgun depending on how the returned string would be modified. The safest way would be to duplicate a const string if the string needs to be stored longer than Lua would keep that pointer valid.

**Comment by axdank** [comment]
@natecraddock How about these changes? I removed the use of @castConst, so if any structure needs a mutable string field, a copy of it is created.

File: src/lib.zig [hunk 1]
```diff
@@ -3196,10 +3196,22 @@ pub const Lua = opaque {
...
 
-                    if (info.sentinel == null) {
-                        return string[0..end];
+                    if (!info.is_const) {
+                        if (!allow_alloc) {
+                            @compileError("toAny cannot allocate memory, try using toAnyAlloc");
+                        }
+
+                        if (info.sentinel != null) {
+                            var buf = try a.?.allocSentinel(u8, end, 0);
```
Comment (natecraddock) [inline src/lib.zig:3205]: Could `a.?.dupeZ()` work here?

**Comment by natecraddock** [comment]
Thank you for working on this!

**Pull request merged**